trio/_tests/test_threads.py

```diff
@@ -810,25 +809,28 @@ def test_from_thread_run_during_shutdown():
     save = []
     record = []
 
-    async def agen():
+    async def agen(token):
         try:
             yield
         finally:
             with pytest.raises(_core.RunFinishedError), _core.CancelScope(shield=True):
-                await to_thread_run_sync(from_thread_run, sleep, 0)
+                await to_thread_run_sync(
+                    partial(from_thread_run, sleep, 0, trio_token=token)
+                )
             record.append("ok")
 
-    async def main():
-        save.append(agen())
+    async def main(use_system_task):
+        save.append(agen(_core.current_trio_token() if use_system_task else None))
         await save[-1].asend(None)
 
-    _core.run(main)
+    _core.run(main, True)  # System nursery will be closed and raise RunFinishedError
+    _core.run(main, False)  # host task will not be rescheduled
     assert record == ["ok"]
```

Comment:
Can you explain why this only produces a single `"ok"` rather than the two that I would expect from looking at it?

Reply:
Past me did not do a great job of documenting this, and possibly didn't understand what was going on at all.  Present me notices that `pytest.raises` is not finding the error it expected, but that is not propagating out to the test runner.

I'll change it so `RunFinishedError` appends one string and no exception appends a different one.